### PR TITLE
fix(node): events.once with generic sign

### PR DESCRIPTION
**Pull request by imcotton**
## Note:
This is following up my previously PR #35130, which would be better have had this addition.

## Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

## Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

## If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

## If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/node/events.d.ts [hunk 1]
```diff
@@ -2,7 +2,8 @@ declare module "events" {
     class internal extends NodeJS.EventEmitter { }
 
     namespace internal {
-        function once(emitter: EventEmitter, event: string | symbol): Promise<any[]>;
+        // tslint:disable-next-line no-unnecessary-generics
+        function once<T extends any[]>(emitter: EventEmitter, event: string | symbol): Promise<T>;
```
Comment (SimonSchick) [inline types/node/events.d.ts]: We shouldn't do that, a type assertion should be used instead.
See https://github.com/Microsoft/dtslint/blob/master/docs/no-unnecessary-generics.md
Eg. a generic should not be used if it only has 1 reference.

Comment (imcotton) [reply types/node/events.d.ts]: Thanks for the reviewing (again), I did saw this alert from `dtslint` and then read this [comment][1] from the issue, do you think it's could be a valid case for marking as `tslint ignore`, or shall letting user to use type assertion as the only way?

[1]: https://github.com/Microsoft/dtslint/issues/76#issuecomment-364201501

Comment (imcotton) [reply types/node/events.d.ts]: /ping @SimonSchick

Comment (SimonSchick) [reply types/node/events.d.ts]: github must've eaten my reply.
And no you should use casting since there is only a single reference to `T.

**Comment by imcotton** [comment]
Thanks for the feedback, closing the PR.

**Pull request closed without merging**